### PR TITLE
Merge minimal with helm install and simpify install paths

### PR DESCRIPTION
**Pull request by frankbu**

### CODE REVIEW

File: content/blog/2018/incremental-traffic-management/index.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Fortunately, a standard Istio deployment already includes a [Gateway](/docs/conc
 
 ## In action: traffic routing with Istio
 
-A simple way to see this type of approach in action is to first setup your Kubernetes environment using the [Platform Setup](/docs/setup/kubernetes/platform-setup/) instructions, and then install Istio using [Helm](/docs/setup/kubernetes/install/minimal/), including only the traffic management components (ingress gateway, egress gateway, Pilot). The following example uses [Google Kubernetes Engine](https://cloud.google.com/gke).
+A simple way to see this type of approach in action is to first setup your Kubernetes environment using the [Platform Setup](/docs/setup/kubernetes/platform-setup/) instructions, and then install the **minimal** Istio profile using [Helm](/docs/setup/kubernetes/install/helm/), including only the traffic management components (ingress gateway, egress gateway, Pilot). The following example uses [Google Kubernetes Engine](https://cloud.google.com/gke).
```
Comment (linsun) [inline content/blog/2018/incremental-traffic-management/index.md]: hmm, this is not accurate, the minimal profile only contains pilot by default.

File: content/docs/concepts/policies-and-telemetry/index.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ At a high level, Mixer provides:
 Beyond these purely functional aspects, Mixer also has [reliability and scalability](#reliability-and-latency) benefits as outlined below.
 
 Policy enforcement and telemetry collection are entirely driven from configuration.
-It's possible to [completely disable these features](/docs/setup/kubernetes/install/minimal/)
+It's possible to completely disable these features by installing the [minimal Istio profile](/docs/setup/kubernetes/install/helm/)
```
Comment (linsun) [inline content/docs/concepts/policies-and-telemetry/index.md]: note: mixer policy is off by default.  we don't necessarily need to refer users to minimal profile here.

File: content/docs/setup/kubernetes/_index.md [hunk 3]
```diff
@@ -46,15 +46,15 @@ We recommend you install Istio for production using the
...
-- [Alibaba Cloud Kubernetes Container Service](/docs/setup/kubernetes/install/alicloud/)
-- [Google Kubernetes Engine](/docs/setup/kubernetes/install/gke/)
-- [IBM Cloud](/docs/setup/kubernetes/install/ibm/)
+- [Alibaba Cloud Kubernetes Container Service](/docs/setup/kubernetes/install/platform/alicloud/)
+- [Google Kubernetes Engine](/docs/setup/kubernetes/install/platform/gke/)
+- [IBM Cloud](/docs/setup/kubernetes/install/platform/ibm/)
 
-If you want to install and use Istio through a Container Network Interface
-(CNI), visit our [CNI guide](/docs/setup/kubernetes/install/cni/).
+If you want your installation to use Istio's Container Network Interface
```
Comment (linsun) [inline content/docs/setup/kubernetes/_index.md]: Should we add a note why someone should look into CNI?  e.g. requiring Istio users to have elevated Kubernetes netadmin permissions is a problem for you, then check out CNI.

File: content/docs/setup/kubernetes/install/helm/index.md [hunk 4]
```diff
@@ -10,11 +10,27 @@ aliases:
...
+customization of the Istio control plane and of the sidecars for the Istio data plane.
+You can start with one of Istio's built-in configuration profiles and then further customize
+the configuration for your specific needs.
+The following built-in configuration profiles are currently available:
+
+1. **minimal**: the minimal set of components necessary to use Istio's [traffic management](/docs/tasks/traffic-management/) features.
+
+1. **demo**: configuration suitable to run the [Bookinfo](/docs/examples/bookinfo/) application and associated tasks.
+    This is the same configuration that is installed with the [Quick Start](/docs/setup/kubernetes/install/kubernetes/) instructions, only using helm has the advantage
+    that you can more easily enable additional features if you wish to explore more advanced tasks.
```
Comment (linsun) [inline content/docs/setup/kubernetes/install/helm/index.md]: I love putting these profiles together!!

File: content/docs/setup/kubernetes/install/helm/index.md [hunk 5]
```diff
@@ -10,11 +10,27 @@ aliases:
...
+
+1. **default**: enables components according to the default [Installation Options](/docs/reference/config/installation-options/)
+    (recommend for production deployments).
+
```
Comment (linsun) [inline content/docs/setup/kubernetes/install/helm/index.md]: should we add the remote profile?  e.g. values-istio-remote.yaml

Comment (linsun) [reply content/docs/setup/kubernetes/install/helm/index.md]: and sds enabled profile?  values-istio-sds-auth.yaml

**Comment by linsun** [comment]
lgtm left a few minor comments, good work @frankbu

File: content/docs/setup/kubernetes/install/helm/index.md (hunk 5 above)
Comment (frankbu) [reply content/docs/setup/kubernetes/install/helm/index.md]: I think so, yes, but in a future PR (probably after 1.1) when we have a good inventory of what all the profiles are, like the table requested in #3460

File: content/blog/2018/incremental-traffic-management/index.md (hunk 1 above)
Comment (frankbu) [reply content/blog/2018/incremental-traffic-management/index.md]: So this is a separate issue from this PR. The blog needs to be updated.

^^^ @crcsmnky

File: content/docs/setup/kubernetes/_index.md (hunk 3 above)
Comment (frankbu) [reply content/docs/setup/kubernetes/_index.md]: The link, just below, explains why.

File: content/docs/concepts/policies-and-telemetry/index.md (hunk 2 above)
Comment (frankbu) [reply content/docs/concepts/policies-and-telemetry/index.md]: I'll reword this part.

**Pull request merged**